tests/test_queuing.py

```diff
@@ -0,0 +1,368 @@
+
+        op1 = qml.PauliZ(0)
+        op2 = qml.PauliZ(1)
+        assert not mock_queuing_context.queue
+
+        with mock_queuing_context:
+            QueuingContext.append(op1)
+
+            assert len(mock_queuing_context.queue) == 1
+            assert op1 in mock_queuing_context.queue
+
+            QueuingContext.remove(op2)
+
+        assert len(mock_queuing_context.queue) == 1
+        assert op1 in mock_queuing_context.queue
+
+    def test_append_multiple_queues(self, three_mock_queuing_contexts):
+        """Test that append appends the operator to multiple queues."""
+
+        op = qml.PauliZ(0)
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+        with three_mock_queuing_contexts[0]:
+            with three_mock_queuing_contexts[1]:
+                with three_mock_queuing_contexts[2]:
+                    QueuingContext.append(op)
+
+        assert len(three_mock_queuing_contexts[0].queue) == 1
+        assert op in three_mock_queuing_contexts[0].queue
+
+        assert len(three_mock_queuing_contexts[1].queue) == 1
+        assert op in three_mock_queuing_contexts[1].queue
+
+        assert len(three_mock_queuing_contexts[1].queue) == 1
+        assert op in three_mock_queuing_contexts[1].queue
+
+    def test_remove_operator_multiple_queues(self, three_mock_queuing_contexts):
+        """Test that remove_operator removes the operator from the queue."""
+
+        op = qml.PauliZ(0)
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+        with three_mock_queuing_contexts[0]:
+            with three_mock_queuing_contexts[1]:
+                with three_mock_queuing_contexts[2]:
+                    QueuingContext.append(op)
+
+                    assert len(three_mock_queuing_contexts[0].queue) == 1
+                    assert op in three_mock_queuing_contexts[0].queue
+
+                    assert len(three_mock_queuing_contexts[1].queue) == 1
+                    assert op in three_mock_queuing_contexts[1].queue
+
+                    assert len(three_mock_queuing_contexts[2].queue) == 1
+                    assert op in three_mock_queuing_contexts[2].queue
+
+                    QueuingContext.remove(op)
+
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+
+class TestQueue:
+    """Test the Queue class."""
+
+    def test_arbitrary_obj(self):
+        """Tests that arbitrary objects can be appended to and removed from the queue."""
+
+        objs = [5, "hi", 1.2, np.einsum, lambda x: x + 1]
+        with qml._queuing.Queue() as q:
+            for obj in objs:
+                q.append(obj)
```

Comment:
Almost makes me want a `Queue.extend()` method 🤔 

In fact, Queue could easily just inherit from sequence and you get all these things for free (similar to the `Wires` class).

Reply:
so it would be a mixin with `QueuingContext` and `Sequence`? Probably worth considering when we flesh out more structured queues